### PR TITLE
remove lint precommit and github actions

### PR DESCRIPTION
**Pull request by amanzanero**
I configured an alias `git pr` using 

```
git config --local alias.pr '!./gradlew ktlintFormat && git add -A && git commit -m "lint" && gt upstack submit'
```

and used it to create this PR!

### CODE REVIEW

**Comment by amanzanero** [comment]
Current dependencies on/for this PR:
* main
  * **PR #17** <a href="https://app.graphite.dev/github/pr/sebvil/mango/17" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sebvil/mango/17?utm_source=stack-comment).

File: app/src/main/java/com/sebastianvm/mango/ui/mvvm/BaseViewModel.kt [hunk 1]
```diff
@@ -10,7 +10,7 @@ interface State
 interface UserAction
 
 abstract class BaseViewModel<S : State, A : UserAction>(
-    initialState: S,
+    initialState: S
```
Comment (amanzanero) [inline app/src/main/java/com/sebastianvm/mango/ui/mvvm/BaseViewModel.kt:13]: lol are our versions of ktlint battling it out for the trailing comma? 😂

Comment (sebvil) [reply app/src/main/java/com/sebastianvm/mango/ui/mvvm/BaseViewModel.kt:13]: No, mine remove it too, but I hadn't committed it (hence this fix lol)

**Review by sebvil — APPROVED** [review]
Lgtm

**Pull request merged**